src/pkg/packager/lint/lint.go

```diff
@@ -25,120 +22,109 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
-// ZarfSchema is exported so main.go can embed the schema file
-var ZarfSchema embed.FS
-
-func getSchemaFile() ([]byte, error) {
-	return ZarfSchema.ReadFile("zarf.schema.json")
+// FileLoader is an interface for reading files, it decouples the lint package from the go embed package
+type FileLoader interface {
+	ReadFile(path string) ([]byte, error)
 }
 
-// Validate validates a zarf file against the zarf schema, returns *validator with warnings or errors if they exist
-// along with an error if the validation itself failed
-func Validate(ctx context.Context, createOpts types.ZarfCreateOptions) (*Validator, error) {
-	validator := Validator{}
-	var err error
+// ZarfSchema is exported so main.go can embed the schema file
+var ZarfSchema FileLoader
 
-	if err := utils.ReadYaml(filepath.Join(createOpts.BaseDir, layout.ZarfYAML), &validator.typedZarfPackage); err != nil {
+// Validate validates a zarf file
+func Validate(ctx context.Context, createOpts types.ZarfCreateOptions) ([]types.PackageError, error) {
```

Comment:
Instead of returning a custom error struct directly in our functions, how about we make our custom error types implement the `error` interface? Here’s why I think this would be beneficial:

1. This lets us add more context to our errors while still returning them as standard error types.

1. It ensures our errors play nicely with Go's built-in error handling functions like `errors.Is`, `errors.As`, and error wrapping with `%w` in `fmt.Errorf`.

1. Handling all errors as the error interface would allow us to manage both third-party and custom errors uniformly, which simplifies the code.